dissect/target/plugins/apps/container/docker.py

```diff
@@ -35,11 +49,62 @@
     ],
 )
 
-DOCKER_NS_REGEX = re.compile(r"\.(?P<nanoseconds>\d{7,})(?P<postfix>Z|\+\d{2}:\d{2})")
 
+DockerLogRecord = TargetRecordDescriptor(
+    "apps/containers/docker/log",
+    [
+        ("datetime", "ts"),
+        ("string", "container"),
+        ("string", "stream"),
+        ("string", "message"),
+    ],
+)
+
+
+def find_installs(target: Target) -> Iterator[Path]:
```

Comment:
I wanted to suggest some changes to the `find_installs()` function, but couldn't due to some deleted lines. Please see the suggestions in a code-block below:
```python
def get_data_path(path: Path) -> Optional[str]:
    """Returns the configured Docker daemon data-root path."""
    try:
        config = json.loads(path.open("rt").read())
    except json.JSONDecodeError as e:
        log.warning("Could not read JSON file '%s'", path)
        log.debug(exc_info=e)

    return config.get("data-root")


def find_installs(target: Target) -> Iterator[Path]:
    """Attempt to find additional configured and existing Docker daemon data-root folders.

    References:
        - https://docs.docker.com/config/daemon/
    """

    default_config_paths = [
        # Linux
        "/etc/docker/daemon.json",
        "/var/snap/docker/current/config/daemon.json",
        # Windows
        "sysvol/ProgramData/docker/config/daemon.json",
    ]

    user_config_paths = [
        # Docker Desktop (macOS/Windows/Linux)
        ".docker/daemon.json",
    ]

    if (default_root := target.fs.path("/var/lib/docker")).exists():
        yield default_root

    for path in default_config_paths:
        if (config_file := target.fs.path(path)).exists():
            if (data_root_path := target.fs.path(get_data_path(config_file))).exists():
                yield data_root_path

    for path in user_config_paths:
        for user_details in target.user_details.all_with_home():
            if (config_file := user_details.home_path.joinpath(path)).exists():
                if (data_root_path := target.fs.path(get_data_path(config_file))).exists():
                    yield data_root_path
```